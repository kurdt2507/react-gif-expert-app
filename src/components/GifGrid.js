import React from 'react';
import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({ category }) => {
    const { data:imagenes, loading } = useFetchGifs(category);

    return (
        <>
            <h3>{ category }</h3>

            { loading && <p>Loading...</p>}

            <div className="card-grid animate__animated animate__fadeIn">
                { imagenes.map((img) => (
                    <GifGridItem
                        key={ img.id }
                        { ...img }
                    />
                )) }
            </div>
        </>
    );
}
