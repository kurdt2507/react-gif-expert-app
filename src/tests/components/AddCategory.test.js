import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import { AddCategory } from '../../components/AddCategory';

describe('Pruebas sobre <AddCategory />', () => {
    const setCategories = jest.fn();
    let wrapper = shallow(<AddCategory setCategories={ setCategories } />);

    beforeEach(() => {
        jest.clearAllMocks();
        wrapper = shallow(<AddCategory setCategories={ setCategories } />);
    });

    test('Valida que el componente coincida con el snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    });
    
    test('Valida el change en el input', () => {
        const input = wrapper.find('input');
        const value = 'Kioña';

        input.simulate('change', { target: { value } });

        expect(wrapper.find('p').text().trim()).toBe(value);
    });
    
    test('No debe hacer el submit cuando el texto es menor o igual a 2 caracteres', () => {
        wrapper.find('form').simulate('submit', {
            preventDefault: () => {}
        });

        expect(setCategories).not.toHaveBeenCalled();
    });
    
    test('Debe llamar setCategories y limpiar la caja de texto', () => {
        const input = wrapper.find('input');
        const value = 'Kioña';

        input.simulate('change', { target: { value } }); // Simula el input change
        // Simula el submit del formulario
        wrapper.find('form').simulate('submit', {
            preventDefault: () => {}
        });

        expect(setCategories).toHaveBeenCalledTimes(1); // Verifica que se haya llamado 1 vez
        expect(setCategories).toHaveBeenCalledWith(expect.any(Function)); // Valida que se haya llamado con una función
        expect(input.prop('value')).toBe('');
    });
    
});
