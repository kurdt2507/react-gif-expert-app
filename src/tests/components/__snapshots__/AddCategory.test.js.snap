// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas sobre <AddCategory /> Valida que el componente coincida con el snapshot 1`] = `
<form
  onSubmit={[Function]}
>
  <p />
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
